src/opm/input/eclipse/Schedule/MSW/Valve.cpp

```diff
@@ -134,8 +162,11 @@ namespace Opm {
         return m_con_flow_coeff;
     }
 
-    double Valve::conCrossArea() const {
-        return m_con_cross_area;
+    double Valve::conCrossArea(const std::optional<const ValveUDAEval>& uda_eval_optional) {
+        m_con_cross_area_value = uda_eval_optional.has_value() ?
+                                    uda_eval_optional.value().value(m_con_cross_area, m_udq_default) :
+                                    m_con_cross_area.getSI();
```

Comment:
Maybe instead of making the member function be non-`const` we can mark the `m_con_cross_area` data member be `mutable` instead?  That way we can still call `conCrossArea()` on a `const Valve` object and we might not have to provide read/write access to the `Valve` and segment set objects.

Reply:
Ah, that's much better, thanks. Will update.